tests/behat/context/ApiContext.php

```diff
@@ -1635,6 +1650,256 @@ public function iShouldGetTheStoredJsonObject(string $name): void
+
+    foreach ($expected_programs as $expected_key => $expected_program)
+    {
+      foreach ($returned_programs as $returned_key => $returned_program)
+      {
+        if ($expected_program['Name'] === $returned_program['name'])
+        {
+          unset($expected_programs[$expected_key], $returned_programs[$returned_key]);
+
+          break;
+        }
+      }
+    }
+    unset($expected_program, $returned_program);
+
+    if (count($returned_programs) > 0)
+    {
+      $values = [];
+      foreach ($returned_programs as $key => $value)
+      {
+        $values[] = $value['name'];
+      }
+      $projects = implode(',', $values);
+      Assert::assertEquals(count($returned_programs), 0, 'Returned projects have ['.$projects.'] unmatched');
+    }
+    if (count($expected_programs) > 0)
+    {
+      $values = [];
+      foreach ($expected_programs as $key => $value)
+      {
+        $values[] = $value['name'];
+      }
+      $projects = implode(',', $values);
+      Assert::assertEquals(count($expected_programs), 0, 'Expected projects have ['.$projects.'] unmatched');
+    }
+  }
+
+  /**
+   * @Then /^The response should contain the following project:$/
+   */
+  public function responseShouldContainTheFollowingProject(TableNode $table): void
+  {
+    $response = $this->getKernelBrowser()->getResponse();
+
+    $returned_programs = json_decode($response->getContent(), true);
+    $expected_programs = $table->getHash();
+
+    Assert::assertEquals(count($returned_programs), count($expected_programs), 'Number of returned programs should be '.count($returned_programs));
+
+    foreach ($expected_programs as $expected_key => $expected_program)
+    {
+      foreach ($returned_programs as $returned_key => $returned_program)
+      {
+        if ($expected_program['Name'] === $returned_program['name'])
+        {
+          unset($expected_programs[$expected_key], $returned_programs[$returned_key]);
+
+          break;
+        }
+      }
+    }
+    unset($expected_program, $returned_program);
+
+    if (count($returned_programs) > 0)
+    {
+      $values = [];
+      foreach ($returned_programs as $key => $value)
+      {
+        $values[] = $value['name'];
+      }
+      $projects = implode(',', $values);
+      Assert::assertEquals(count($returned_programs), 0, 'Returned projects have ['.$projects.'] unmatched');
+    }
+    if (count($expected_programs) > 0)
+    {
+      $values = [];
+      foreach ($expected_programs as $key => $value)
+      {
+        $values[] = $value['name'];
+      }
+      $projects = implode(',', $values);
+      Assert::assertEquals(count($expected_programs), 0, 'Expected projects have ['.$projects.'] unmatched');
+    }
+  }
+
+  /**
+   * @Then /^The response should contain projects in the following order:$/
+   */
+  public function responseShouldContainProjectsInTheFollowingOrder(TableNode $table): void
```

Comment:
All those new step definitions are quite nice since they highly reduce the amount of necessary test code.
But, and that is a big one. Now we are not testing with a way lower precision.  So if we are going to keep them there is still much work to do:

Especially in the API, we must not check that projects name are returned in the correct order. Every field must be checked for the correct content. Also, it must be checked that all fields are returned and nothing more. 
Therefore I'm not even sure if it would be enough to compare object after the json_decode.

In my opinion, we should not lower the testing precision concerning the API. Minor changes we are not tracking could result in broken code by other teams using our API. To immediately see changed behavior, I see no other possibility than to compare the whole JSON response with at least an expected JSON response based on the projects (It can still be generated). 

The same applies to all these step definitions.